crates/bevy_ecs/src/reflect/entity_commands.rs

```diff
@@ -189,15 +189,18 @@ fn insert_reflect(
     type_registry: &TypeRegistry,
     component: Box<dyn Reflect>,
 ) {
-    let type_info = component.reflect_type_path();
+    let type_path = component
+        .get_represented_type_info()
+        .expect("component should represent a type.")
+        .type_path();
```

Comment:
One last suggestion. Let's split into two variables: `type_info` and `type_path`.